src/operators/pipe.ml

```diff
@@ -236,18 +239,6 @@ class pipe ~kind ~replay_delay ~data_len ~process ~bufferize ~log_overfull ~max
                 true
             | _, `Nothing -> restart)
 
-    method private child_tick =
-      (Clock.get source#clock)#end_tick;
-      source#after_output
-
-    (* See smactross.ml for details. *)
-    method private set_clock =
-      let child_clock = Clock.create_known (new Clock.clock self#id) in
-      Clock.unify self#clock
-        (Clock.create_unknown ~sources:[] ~sub_clocks:[child_clock]);
-      Clock.unify child_clock source#clock;
-      Gc.finalise (fun self -> Clock.forget self#clock child_clock) self
-
```

Comment:
There is really no need for explicit clock handling anymore, really?

Reply:
Same thing, checkout this file: https://github.com/savonet/liquidsoap/blob/40057dce0da8f54eec4b6af0ca951aa3ce635323/src/tools/child_support.ml